README.md

```diff
@@ -17,76 +16,126 @@ _A table of contents to help people find things_
 
 ```js
-document.addEventListener('DOMContentLoaded', function() {
-	document.dispatchEvent(new CustomEvent('o.DOMContentLoaded'));
+const oMessage = require('o-message');
+const importantMessage = new oMessage(null, {
+	type: 'alert-bleed',
+	status: 'error',
+	content: {
+		highlight: 'Something has gone wrong.'
+		detail: 'The quick brown fox did not jump over the lazy dogs.'
+	}
 });
 ```
 
-### Sass
-_Remember to start your codeblocks with three backticks and "sass" so your markup is syntax highlighted correctly._
-
-_Though it's not practical to repeat every aspect of Origami modules convention for every component, **A LOT** of people get tripped up by silent mode, so this line (remember to change the o-component-boilerplate to your component name) is useful if you have Sass:_
-
-As with all Origami components, o-component-boilerplate has a [silent mode](http://origami.ft.com/docs/syntax/scss/#silent-styles). To use its compiled CSS (rather than using its mixins with your own Sass) set `$o-component-boilerplate-is-silent : false;` in your Sass before you import the o-component-boilerplate Sass.
-
-## Troubleshooting
-_This is a good place to put problems that come up repeatedly_
-
-### The thing with the thing isn't working
-Fix it by turning it off and on again
+#### Options
+`o-message` allows for several configuration options that will change the type of message and its visual styling.
+
+The only required options are listed in the example _above_. These are:
+- `type`: String. The o-message variant. The available variants are 'alert', 'alert-bleed' and 'alert-inner'.
+- `status`: String. Alert variants require a status, and the options are 'success', 'neutral' and 'error'.
+- `content`: Object. Holds the following values for text properties:
+	- `highlight`: String. The highlighted text in a message. Defaults to `null`
+	-	`detail`: String. The detail about the nature of a message.
+
+The following options are not required, and all have a default value:
+
+- `autoOpen`: Boolean. Whether to open the message automatically, defaults to `true`.
+- `messageClass`: String. The base class name for the component's elements, defaults to `o-message`.
+- `variant`: String. The message variant which allows a message to be a bleed message or an inline message. Defaults to `null`.
+- `parentElement`: String. This determines the element that the message will be appended to. If none is declared, it will automatically append to the body, or an element with the data attribute `data-o-component=o-message`, defaults to `null`.
+- `content`: Object. Holds the following values for text properties:
+	-	`additionalInfo`: String. More information about the message –  only applies to an `alert-inner` message. Defaults to `null`
+- `button`: Object. Holds the following values for button properties:
+	- `text`: String. text value of the button.
+	- `url`: String. The URL the button links to.
+- `link`: Object. Holds the following values for link properties:
```

Comment:
`button` and `link` are not actually for buttons or links specifically in the html/semantic sense...  Something like`primary-action` `secondary-action`might be more appropriate. Or more scalable but probably overkill 
```
actions: [
    primary: [
        text,
        url
    ],
    [...]
]
```
?

Reply:
It may be overkill, but I think its worth being more explicit.